### PR TITLE
[CoW] Migrate Eth Flow Orders

### PR DESCRIPTION
**Pull request by bh2smith**
Migrating to Dunesql.

### CODE REVIEW

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql [hunk 1]
```diff
@@ -23,34 +24,38 @@ eth_flow_orders as (
...
-        -- OrderHash returned by createOrder with excluded fix values (owner = contract_address, validTo = max u32)
-        -- https://github.com/cowprotocol/ethflowcontract/blob/9c74c8ba36ff9ff3e255172b02454f831c066865/src/CoWSwapEthFlow.sol#L81-L84
-        concat(output_orderHash, substring(event.contract_address, 3, 40), 'ffffffff') as order_uid
-    from {{ source('cow_protocol_ethereum', 'CoWSwapEthFlow_evt_OrderPlacement') }} event
-    inner join {{ source('cow_protocol_ethereum', 'CoWSwapEthFlow_call_createOrder') }} call
+        date_format(
+            from_unixtime(bytearray_to_decimal(from_hex(substring(cast(data as varchar), 19, 8)))),
+           '%Y-%m-%d %T'
+        ) AS valid_to,
+      substring(cast(data as varchar), 3, 16) as quote_id_hex,
```
Comment (jkylling) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:36]: We try to prefer using varbinary for a binary data. Could this be changed to `bytearray_substring(data, 1, 8)`?

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql [hunk 2]
```diff
@@ -23,34 +24,38 @@ eth_flow_orders as (
...
+      bytearray_to_decimal(from_hex(substring(cast(data as varchar), 3, 16))) as quote_id,
```
Comment (jkylling) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:37]: Same here. In general it should never be necessary to cast varbinary to varchar, unless the end result is a varchar which will be part of some larger string.

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql [hunk 3]
```diff
@@ -23,34 +24,38 @@ eth_flow_orders as (
...
+      JSON_EXTRACT_SCALAR(event."order", '$.sellAmount') as sell_amount,
+      JSON_EXTRACT_SCALAR(event."order", '$.feeAmount') as fee_amount,
+      JSON_EXTRACT_SCALAR(event."order", '$.buyAmount') as buy_amount,
+      JSON_EXTRACT_SCALAR(event."order", '$.buyToken') as buy_token,
+      JSON_EXTRACT_SCALAR(event."order", '$.receiver') as receiver,
+      JSON_EXTRACT_SCALAR(event."order", '$.appData') as app_hash,
```
Comment (jkylling) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:43]: Can these be casted to explicit types?

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql [hunk 4]
```diff
@@ -23,34 +24,38 @@ eth_flow_orders as (
...
+      -- OrderHash returned by createOrder with excluded fix values (owner = contract_address, validTo = max u32)
+      -- https://github.com/cowprotocol/ethflowcontract/blob/9c74c8ba36ff9ff3e255172b02454f831c066865/src/CoWSwapEthFlow.sol#L81-L84
+      concat(
```
Comment (jkylling) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:46]: Prefer `bytearray_concat`

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql (hunk 1 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:36]: Ya that was just missed- thanks.

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql (hunk 2 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:37]: We are slicking a string here. Is there some way of slicing varbinary

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql (hunk 3 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:43]: oh yes, that was definitely intended.

File: models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql (hunk 2 above)
Comment (jkylling) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_eth_flow_orders.sql:37]: This line can be replaced by
```
bytearray_to_decimal(bytearray_substring(data, 1, 8)) as quote_id,
```
see https://dune.com/queries/2720836

**Comment by bh2smith** [comment]
Thank @jkylling for your careful review. I have addressed and resolved all the type issues above. PoC Query has also been updated.

https://dune.com/queries/2715069

**Review by jeff-dude — APPROVED** [review]
thank you 🙌

**Pull request merged**